src/buffer.c

```diff
@@ -38,6 +38,55 @@ BUFFER_HANDLE BUFFER_new(void)
     return (BUFFER_HANDLE)temp;
 }
 
+// return 0 if safemalloc succeeds else return non-zero
+int BUFFER_safemalloc(BUFFER_HANDLE* handleptr, size_t size)
+{
+    int result;
+    // if handleptr or handle is NULL, return immediately
+    if (!handleptr || !(*handleptr))
+    {
+        result = __LINE__;
+    }
+    else
+    {
```

Comment:
No need to check these as this should be internal and these should already be checked.